fe/fe-core/src/main/java/com/starrocks/sql/optimizer/Optimizer.java

```diff
@@ -280,6 +292,19 @@ private OptExpression logicalRuleRewrite(ConnectContext connectContext,
         rootTaskContext.setRequiredColumns(requiredColumns.clone());
         ruleRewriteOnlyOnce(tree, rootTaskContext, RuleSetType.PRUNE_COLUMNS);
 
+        if (sessionVariable.isEnableRboTablePrune()) {
+            ruleRewriteOnlyOnce(tree, rootTaskContext, RuleSetType.PARTITION_PRUNE);
+            ruleRewriteIterative(tree, rootTaskContext, new MergeProjectWithChildRule());
+            tree = new ReorderJoinRule().rewrite(tree, context);
+            tree = new SeparateProjectRule().rewrite(tree, rootTaskContext);
```

Comment:
`Projection` never appear here

Reply:
ReorderJoinRule requires that merge projection into its child(MergeProjectWithChildRule), but rules following ReorderJoinRule requires LogicalProjectOperator is separated from operator as a standalone operator.